src/ValidationMessage/index.jsx

```diff
@@ -0,0 +1,85 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+
+import newId from '../utils/newId';
+import styles from './ValidationMessage.scss';
+
+const inputProps = {
+  dangerIconDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  className: PropTypes.arrayOf(PropTypes.string),
+  errorId: PropTypes.string,
+  isValid: PropTypes.bool,
+  themes: PropTypes.arrayOf(PropTypes.string),
+  validationMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+};
+
+const defaultProps = {
+  dangerIconDescription: '',
+  className: [],
+  errorId: undefined,
+  isValid: true,
+  themes: [],
+  validationMessage: '',
+};
+
+class ValidationMessage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const id = this.props.errorId ? this.props.errorId : `error-${newId('validationMessage')}`;
+    this.state = {
+      errorId: id,
+    };
+  }
+
+  hasDangerTheme() {
+    return this.props.themes.indexOf('danger') >= 0;
+  }
+
+  render() {
+    return (
+      <div
+        className={classNames(
+          styles['invalid-feedback'],
+          { [styles['invalid-feedback-nodanger']]: !this.hasDangerTheme() },
+          this.props.className,
+        )}
+        id={this.state.errorId}
+        aria-live="polite"
+      >
+        { this.props.isValid ? (
+          <span />
+        ) : [
+          (this.hasDangerTheme() &&
+          <span key="0">
+            <span
+              className={classNames(
+                FontAwesomeStyles.fa,
+                FontAwesomeStyles['fa-exclamation-circle'],
+                styles['fa-icon-spacing'],
+              )}
+              aria-hidden
+            />
+            <span
+              className={classNames(styles['sr-only'])}
+            >
+              {this.props.dangerIconDescription}
+            </span>
+          </span>
+          ),
+          <span key="1">
```

Comment:
Since `React` uses `key`s to determine if something has changed if the `this.props.validationMessage` changes will this not be reflected in subsequent renders?

Reply:
I'm going to switch this code over to wrap the list of elements in `<React.Fragment>`s instead to avoid this `key`s issue.

Reply:
That will make this component incompatible with anything before `React` 16 right?

Is this something we need to be concerned about? (Probably not)

Reply:
I'm not aware of anything that's using Paragon and not on React 16 so I don't think it is something we should be concerned about.